### ISSUE TITLE
`ver-bump` is not on Homebrew

### ISSUE THREAD

**Issue by jv-k** [jv-k/bump-version.sh#24]
**Status:** scheduled for 2.0.0.
**Plan:** [`docs/v2.0-release-plan.md` § M6 — Homebrew distribution](../blob/feat/v2.0/docs/v2.0-release-plan.md#m6--homebrew-distribution-resolves-issue-24)

## Summary

Ship `ver-bump` as a Homebrew-installable package so macOS/Linux users don't need `npm`, a manual clone, or an install script.

## Scope

- **Own tap first.** New repo `jv-k/homebrew-ver-bump` with `Formula/ver-bump.rb`. `brew tap jv-k/ver-bump && brew install ver-bump` must succeed on a clean machine.
- Formula depends on `bash`, `jq`.
- `test do` block (required by `brew audit --strict`): at minimum `system "#{bin}/ver-bump", "--help"`.
- CI on the tap repo: `brew audit --strict --online` on every push + tag.
- Release automation: keep tap `url`/`sha256` in sync when `jv-k/ver-bump` publishes a new tag (workflow on the main repo pushing to the tap, or tag-triggered workflow on the tap repo itself).
- Main repo `README.md`: add `brew install jv-k/ver-bump/ver-bump` as a first-class install path.

## Deferred to post-2.0

- `homebrew-core` submission (needs 30-day stability + adoption).
- `basher` distribution — tracked separately.

## Acceptance

- [ ] `brew tap jv-k/ver-bump && brew install ver-bump` succeeds on a clean macOS runner
- [ ] `brew audit --strict --online Formula/ver-bump.rb` passes
- [ ] Tap repo has CI that re-audits on every change
- [ ] Main repo README shows brew install path

**Comment by jv-k** [comment]
> *This was generated by AI during triage.*

## Agent Brief

**Category:** enhancement
**Summary:** Stand up a Homebrew tap (`jv-k/homebrew-ver-bump`) so `brew tap jv-k/ver-bump && brew install ver-bump` works, with audit CI on the tap and automatic formula bumps when new releases are tagged.

**Prerequisites (maintainer, before an agent starts):**

- [x] Create the empty public repo `jv-k/homebrew-ver-bump` (repo creation under the owner's account is not delegated).
- [x] If the chosen sync mechanism pushes cross-repo from `jv-k/ver-bump`, add a secret (fine-grained PAT or deploy key) with write access to the tap repo. Not needed if the automation lives entirely on the tap repo.

**Current behavior:**
Install paths are npm/pnpm global install, manual clone, and the install script. The README's Homebrew section is a "Coming soon" link back to this issue. No tap repo or formula exists anywhere. The latest stable tag is `v2.0.1` and `v2.0.0` is the latest GitHub release, so tag tarballs exist for a formula to point at. ADR-13 deferred this work until 2.0 shipped — that gate has now lifted.

**Desired behavior:**
On a clean macOS machine, `brew tap jv-k/ver-bump && brew install ver-bump` installs a working `ver-bump` command. The formula declares `bash` and `jq` as dependencies, installs the main script as `ver-bump`, and installs the `lib/*.sh` files it sources at runtime such that the brew-installed command resolves them correctly (Homebrew symlinks the bin — the script's lib lookup must survive that). When a new stable tag is published on `jv-k/ver-bump`, automation updates the tap formula's `url` and `sha256` without manual editing (mechanism is the agent's choice; see prerequisites). Pre-release tags (e.g. `-rc.N`) must not trigger formula bumps.

**Key interfaces:**

- `Formula/ver-bump.rb` in the tap repo — Homebrew's required layout. Fields: `desc`, `homepage`, `url` pointing at a version-tag tarball of `jv-k/ver-bump`, `sha256`, MIT `license`, `depends_on "bash"` and `depends_on "jq"`, and a `test do` block (required by `brew audit --strict`) that at minimum runs the installed binary with `--help`. A skeleton exists in the main repo's release-plan doc (§ M6 — Homebrew distribution).
- The main script's library-resolution logic — it sources sibling `lib/*.sh`; verify it resolves them through Homebrew's symlinked bin path. If it doesn't, a minimal fix to resolution (e.g. following symlinks to the real script location) is in scope.
- Tap repo CI — runs `brew audit --strict --online` (and ideally `brew install --build-from-source` + `brew test`) on every push and PR to the tap.
- Main repo README — the existing Homebrew "Coming soon" subsection becomes a first-class install path documenting `brew install jv-k/ver-bump/ver-bump`.

**Acceptance criteria:**

- [ ] `brew tap jv-k/ver-bump && brew install ver-bump` succeeds on a clean macOS runner
- [ ] The installed `ver-bump --help` and `ver-bump --version` run successfully (lib sourcing works through the symlinked bin)
- [ ] `brew audit --strict --online Formula/ver-bump.rb` passes
- [ ] Tap repo CI re-audits the formula on every push/PR
- [ ] Publishing a new stable tag on `jv-k/ver-bump` produces an automated formula `url`/`sha256` bump (commit or PR) on the tap; pre-release tags do not
- [ ] Main repo README documents the brew install path and no longer says "Coming soon"

**Out of scope:**

- `homebrew-core` submission (needs 30-day stability and adoption — future issue)
- basher distribution (#39) and npm publish automation (#95)
- Any behavior change to ver-bump beyond the minimal lib-resolution fix, if one is needed
- Linux-specific (Homebrew-on-Linux) testing beyond what the standard formula setup provides

**Comment by jv-k** [comment]
> *This was generated by AI during triage.*

## Implementation progress

**Done and verified locally:**
- Tap authored at the canonical local tap path (`$(brew --repository)/Library/Taps/jv-k/homebrew-ver-bump`), committed on `main`: formula for **v2.0.1** (sha256 pinned), CI workflow (`brew style` + `brew audit --strict --online` + install + `brew test` on macOS & Linux), scheduled **auto-bump** workflow (checks for a newer stable `vX.Y.Z` tag every 6 h, updates `url`/`sha256`, commits — no cross-repo secret needed; pre-release tags ignored), README, dependabot.
- `brew install` from the local tap succeeds: keg bin symlink → `libexec/ver-bump.sh`, shebang rewritten to Homebrew bash, `lib/*.sh` + `package.json` resolve through the symlink (`realpath`), `ver-bump --version` → `ver-bump 2.0.1`, `--help` exits 0.
- Main-repo README now documents the brew install path — PR #103 (merge once the tap is live).

**Blocked (maintainer):** creating the public repo `jv-k/homebrew-ver-bump` was permission-gated, as anticipated in the brief. Once it exists and the local tap is pushed, CI runs the strict online audit on the clean runners (the local machine couldn't run `brew audit`/`brew test` — Homebrew's vendored-ruby gem build fails there, an environment issue unrelated to the formula).

**Acceptance criteria status:**
- [ ] `brew tap jv-k/ver-bump && brew install ver-bump` on a clean macOS runner — pending repo push + CI
- [x] Installed `--help` / `--version` work through the symlinked bin (verified locally)
- [ ] `brew audit --strict --online` passes — pending CI (local env can't run audit)
- [x] Tap CI re-audits every push/PR — workflow committed
- [x] Stable-tag-only automated `url`/`sha256` bump — workflow committed
- [x] Main repo README brew install path — PR #103

Refs #24.

**Comment by jv-k** [comment]
> *This was generated by AI during triage.*

## Tap is live 🎉

The maintainer created/authorized the public repo and the tap is now published: **https://github.com/jv-k/homebrew-ver-bump** (About, homepage, and topics configured; `homebrew`/`brew`/`version-bump` topics also added to this repo).

What happened since the last update:

- Local tap pushed to `jv-k/homebrew-ver-bump` `main`.
- The **auto-bump workflow proved itself on its first dispatch**: it detected `v2.0.2` (released today, after the formula was authored against 2.0.1), updated `url`/`sha256`, and committed — no cross-repo secret involved. Pre-release filtering untested in anger but logic-reviewed.
- First CI run caught one `brew style` offense (`Homebrew/FormulaPathMethods`: `Formula["bash"].opt_bin` → `formula_opt_bin("bash")`); fixed and pushed.
- CI is now **green on macOS and Linux**: `brew style`, `brew audit --strict --online`, from-source install, and `brew test` all pass.

**Acceptance criteria:**
- [x] Install succeeds on a clean macOS runner (CI install + `brew test` on `macos-latest`)
- [x] Installed `--help` / `--version` work through the symlinked bin
- [x] `brew audit --strict --online Formula/ver-bump.rb` passes
- [x] Tap CI re-audits every push/PR
- [x] Stable-tag-only automated formula bump (verified live: 2.0.1 → 2.0.2)
- [ ] Main repo README documents the brew path — PR #103 is open, but the VerBump rename (#104) is actively rewriting the README; recommend merging or refreshing #103 after the rename sweep lands to avoid conflicting edits.

Remaining for the maintainer: decide the #103 timing, note the tap repo may want renaming alongside #104 (`homebrew-ver-bump` → whatever the brand rename dictates; GitHub redirects make this cheap), and close this issue when satisfied. Two dependabot PRs on the tap predate the style fix and just need a rebase/re-run.

Refs #24.

**Comment by jv-k** [comment]
> *This was generated by AI during triage.*

## Tap restructured for the VerBump 3.0 rebrand

The install spell shortened per maintainer direction, and the formula now tracks the rename:

- Tap repo renamed **`jv-k/homebrew-ver-bump` → `jv-k/homebrew-tap`** (GitHub redirects the old URL). One tap for all jv-k tools from here on.
- Formula renamed **`ver-bump` → `verbump`** and bumped to **v3.0.0**; it installs the command as **`VerBump`**, matching the npm bin entry from the rebrand. Source tarballs still come from `jv-k/ver-bump` (repo and npm package keep the historical name).
- New install spell: `brew install jv-k/tap/verbump` — or `brew tap jv-k/tap && brew install verbump`.
- CI green on macOS + Linux after the restructure (`brew style`, `brew audit --strict --online`, install, `brew test`).
- PR #103 refreshed: merged current `main` into the branch (no force-push) and updated the README section to the new spell and the `VerBump` command name.

`homebrew-core` submission remains the future path to a bare `brew install verbump`.

Refs #24.

**Comment by jv-k** [comment]
> *This was generated by AI during triage.*

## Done — closing per maintainer direction

Final state:

- **PR #103 merged** — README's Homebrew section documents `brew install jv-k/tap/verbump` (conflicts from the 4.0 rename resolved via merge; branch deleted).
- **Formula bumped to v4.0.0**, tracking the full rename (`VerBump.sh` script, `VerBump` command, `jv-k/VerBump` repo). sha256 verified from two independent downloads. Verified locally: fresh `brew install jv-k/tap/verbump` → `VerBump 4.0.0`, `--help` clean.
- Auto-bump workflow handles future stable tags; the script-name change that would have broken it is settled.
- Homebrew-core submission (bare `brew install verbump`) remains future work — blocked on the notability bar (~75 stars; currently 26).

Heads-up for the maintainer, unrelated to brew but discovered en route: **v4.0.0 has a tag but no GitHub release**, and the npm package was renamed `ver-bump` → `verbump` — the OIDC Trusted Publishing setup on npmjs.com is configured for the old package name, so the publish job needs a Trusted Publisher entry for `verbump` before the next release-published event can publish.

Refs #24.
